cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -68,23 +68,17 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
                             @PathParam("namespace-id") String namespaceId) throws Exception {
     checkSourceControlFeatureFlag();
     NamespaceId namespace = validateNamespaceId(namespaceId);
-    RepositoryConfigRequest repoRequest = getRepositoryConfigRequest(request);
-
+    
     try {
-      if (repoRequest == null || repoRequest.getConfig() == null) {
-        throw new RepositoryConfigValidationException("Repository configuration must be specified.");
-      }
-
-      repoRequest.getConfig().validate();
+      RepositoryConfigRequest repoRequest = validateAndGetRepoConfig(request);
       if (repoRequest.shouldTest()) {
-        // TODO: CDAP-20252, add the validate logic once the SourceControlManager module is ready
+        sourceControlService.validateRepository(namespace, repoRequest.getConfig());
```

Comment:
With this call, CDAP will try to validate the Git config by reaching out to the Git hosting service. IIRC, we didn't want to do this during PUT, but have a separate API just for validation. In PUT, we do offline validation without reaching out to remote APIs.

Reply:
Sent the design doc link offline, basically we wanted to use the same endpoint and have a `test` boolean flag to indicate if we validate/set the config

Reply:
Looks good.